### ISSUE TITLE
More 'meta-data' for PageActions or a finished method

### ISSUE THREAD

**Issue by Da-Fecto** [processwire/processwire-requests#77]
### Short description of the enhancement

I was developing a page action to export pages to .xlsx. The Action is based on your PageActionExportCSV.module. I've been utilising the Spout library to write to a xlsx file. 

Spout need to open a file once and close when all batches have run. But so far I'm unable to find a good way to determine when the last item is processed.

What I have done so far is getting the total items to be processed and add a custom counter to get track of the current item. Then when the current item is the last item close the writer so the results will be saved as xlsx.

This is very ugly hack but it looks like it is working.

```
// Get total  
$this->runner = $this->getRunner();
$lister = $this->runner->__get('lister');
$this->total = $lister->sessionGet('knownTotal')['total'];
```

2 possibilities

* I wish there was a method that gets called when the actions are finished. 
* 2 properties to check with the current iteration number and the total number of items.

**Comment by LostKobrakai** [comment]
Isn't the lister using WireAction::executeMultiple()?

**Comment by Da-Fecto** [comment]
yep @LostKobrakai and it receives a batch of items. But the last item in the batch isn't the last item to be processed.
